Source/WebKit/UIProcess/_WKWebViewPrintFormatter.mm

```diff
@@ -96,17 +128,19 @@ - (CGRect)rectForPageAtIndex:(NSInteger)pageIndex
 
 - (void)drawInRect:(CGRect)rect forPageAtIndex:(NSInteger)pageIndex
 {
-    if (!_printedDocument) {
-        _printedDocument = self._webView._printProvider._wk_printedDocument;
-        if (!_printedDocument)
+    CGPDFDocumentRef printedDocument = [self _printedDocument];
+    if (!printedDocument) {
+        [self._webView._printProvider _wk_requestDocumentForPrintFormatter:self];
```

Comment:
Feels like it'd be nice if _wk_requestDocumentForPrintFormatter would return the printed document instead of relying on a side effect from this function

Reply:
I didn't make this change, because the implementation of `_wk_requestDocumentForPrintFormatter:` in the content view relies on the printed document getting set on the formatter as a side effect of async IPC. So, the "side effect" dependency would simply be moved from here into the content view.

Additionally, I would need to expose a `_printedDocument` property on `_WKWebViewPrintFormatter`. This is necessary as the printed document is now stored per-formatter. Finally, there would be duplicate code when returning from both the content view and PDF view cases (reaching back into the formatter).

I named the method `requestDocumentFor...` rather than `documentFor...` to indicate the lack of a return value.